git_machete/cmd.py

```diff
@@ -648,31 +649,128 @@ def annotate(b: str, words: List[str]) -> None:
-    if not org_name_for_github_remote:
+    org_and_repo_for_github_remote: Dict[str, Tuple[str, str]] = {
+        remote: parse_github_remote_url(url) for remote, url in url_for_remote.items() if is_github_remote_url(url)
+    }
+    if not org_and_repo_for_github_remote:
         raise MacheteException(fmt('Remotes are defined for this repository, but none of them corresponds to GitHub (see `git remote -v` for details)'))
 
+    if len(org_and_repo_for_github_remote) == 1:
+        return list(org_and_repo_for_github_remote.items())[0]
+
+    if 'origin' in org_and_repo_for_github_remote:
+        return 'origin', org_and_repo_for_github_remote['origin']
+
+    raise MacheteException(f'Multiple non-origin remotes correspond to GitHub in this repository: '
+                           f'{", ".join(org_and_repo_for_github_remote.keys())}, aborting')
+
+
+def create_github_pr(cli_ctxt: CommandLineContext, head: str, draft: bool) -> None:
+    global annotations, up_branch
+    from git_machete.github import GitHubPullRequest, add_assignees_to_pull_request, add_reviewers_to_pull_request, \
+        create_pull_request, derive_current_user_login, set_milestone_of_pull_request
+
+    base: Optional[str] = up_branch.get(head)
+    if not base:
+        raise MacheteException(f'Branch {head} does not have a parent branch (it is a root), base branch for the PR cannot be established')
+
     org: str
     repo: str
-    if len(org_name_for_github_remote) == 1:
-        org, repo = list(org_name_for_github_remote.values())[0]
-    elif len(org_name_for_github_remote) > 1:
-        if 'origin' in org_name_for_github_remote:
-            org, repo = org_name_for_github_remote['origin']
-        else:
-            raise MacheteException(f'Multiple non-origin remotes correspond to GitHub in this repository: '
-                                   f'{", ".join(org_name_for_github_remote.keys())}, aborting')
+    _, (org, repo) = derive_remote_and_github_org_and_repo(cli_ctxt)
     current_user: Optional[str] = derive_current_user_login()
-    debug(cli_ctxt, 'sync_annotations_to_github_prs()', 'Current GitHub user is ' + (current_user or '<none>'))
+    debug(cli_ctxt, f'create_github_pr({head})', f'organization is {org}, repository is {repo}')
+    debug(cli_ctxt, f'create_github_pr({head})', 'current GitHub user is ' + (current_user or '<none>'))
+
+    def slurp_file_or_empty(path: str) -> str:
+        try:
+            return open(path).read()
+        except IOError:
+            return ''
+
+    title: str = popen_git(cli_ctxt, "log", "-1", "--format=%s").strip()
+    description_path = get_git_subpath(cli_ctxt, 'info', 'description')
+    description: str = slurp_file_or_empty(description_path)
+
+    pr: GitHubPullRequest = create_pull_request(org, repo, head=head, base=base, title=title, description=description, draft=draft)
```

Comment:
If:
* branch is not pushed yet OR
* a PR for the branch already exists,
then the user is left with a cryptic 422 HTTP status, no further details :/

Reply:
Possibly some `traverse`-like interactive behavior for pushing the branch would be useful here